web/cache.go

```diff
@@ -38,20 +38,26 @@ type cacheEntry []byte
 
 func (c *cache) newCacheEntry() cacheEntry {
 	ce := make([]byte, widthPermalink+encoding.WidthTime+1)
-	permalink := uuid.NewRandom()
-	copy(ce[idxPermalink:], permalink)
+	permalink := uuid.New()
+	copy(ce[idxPermalink:], permalink[:])
 	encoding.EncodeTime(ce[idxTime:], time.Now().Add(c.ttl))
 	return ce
 }
 
 func (ce cacheEntry) permalink() string {
-	return uuid.UUID(ce.permalinkBytes()).String()
+	return ce.uuid().String()
 }
 
 func (ce cacheEntry) permalinkBytes() []byte {
 	return ce[idxPermalink : idxPermalink+widthPermalink]
 }
 
+func (ce cacheEntry) uuid() uuid.UUID {
+	var arr [16]byte
+	copy(arr[:], ce.permalinkBytes())
+	return uuid.UUID(arr)
+}
```

Comment:
This is the only other tricky part -- if we do this a lot it might make sense to avoid the copy if possible

Reply:
We don't do it a lot.